apps/dalmatiner_db/src/dalmatiner_tcp.erl

```diff
@@ -76,7 +75,20 @@ loop(Socket, Transport, State) ->
                                         dict = bkt_dict:new(Bucket,
                                                             State#state.n,
                                                             State#state.w)},
+                                {incomplete, <<>>});
+                {stream, Bucket, Delay, Resolution} ->
+                    lager:info("[tcp] Entering stream mode for bucket '~s' "
+                               "with max delay: ~p and resolution: ~pms",
+                               [Bucket, Delay, Resolution]),
+                    ok = dalmatiner_opt:set_resolution(Bucket, Resolution),
+                    ok = Transport:setopts(Socket, [{packet, 0}]),
```

Comment:
Remove the dynamic function call on line 84. Only modules that define callbacks should make dynamic calls.